apps/admin/backend/src/app.ts

```diff
@@ -166,13 +171,81 @@ function buildApi({
       return auth.unprogramCard(constructAuthMachineState(workspace));
     },
 
-    async writeBallotPackageSignatureFile(input: {
-      ballotPackagePath: string;
-    }): Promise<void> {
-      await artifactAuthenticator.writeSignatureFile({
-        type: 'ballot_package',
-        path: input.ballotPackagePath,
+    async saveBallotPackageToUsb(): Promise<Result<void, 'no_usb_drive'>> {
+      await logger.log(LogEventId.SaveBallotPackageInit, 'election_manager');
+
+      const ballotPackageZipStream = new ZipStream();
+      const electionDefinition = getCurrentElectionDefinition(workspace);
+      assert(electionDefinition !== undefined);
+      const systemSettings =
+        store.getSystemSettings() ?? DEFAULT_SYSTEM_SETTINGS;
+      await addFileToZipStream(ballotPackageZipStream, {
+        path: 'election.json',
+        contents: electionDefinition.electionData,
+      });
+      await addFileToZipStream(ballotPackageZipStream, {
+        path: 'systemSettings.json',
+        contents: JSON.stringify(systemSettings, null, 2),
       });
+      ballotPackageZipStream.finalize();
+
+      const tempDirectory = dirSync().name;
+      const ballotPackageFileName = generateFilenameForBallotExportPackage(
+        electionDefinition,
+        new Date()
+      );
+      await fs.writeFile(
+        join(tempDirectory, ballotPackageFileName),
```

Comment:
We're computing this value multiple times. Make a variable?

Reply:
I'd actually done this intentionally because I thought it made the code clearer, e.g.

```ts
await fs.copyFile(
  join(tempDirectory, ballotPackageFileName),
  join(usbBallotPackageDirectory, ballotPackageFileName)
);
```

reads a bit clearer to me than

```ts
await fs.copyFile(
  tempDirectoryBallotPackageFilePath,
  usbBallotPackageFilePath
);
```

But on second thought, they're probably comparable in readability. Can update

Reply:
✅ https://github.com/votingworks/vxsuite/pull/3554/commits/7a4bb362091acadcaa701f1bdcfabf8e13347455

Reply:
Thanks. My thinking was that someone updating it might forget to `join` one of them.